src/Components/Components/src/ComponentBase.cs

```diff
@@ -322,15 +319,18 @@ private static void HandleException(Exception ex)
             Console.Error.WriteLine($"[{ex.GetType().FullName}] {ex.Message}\n{ex.StackTrace}");
         }
 
-        void IHandleEvent.HandleEvent(EventHandlerInvoker binding, UIEventArgs args)
+        Task IHandleStateChange.HandleStateChangeAsync(Task task)
         {
-            var task = binding.Invoke(args);
-            ContinueAfterLifecycleTask(task);
+            var (isAsync, asyncTask) = ProcessLifeCycletask(task);
```

Comment:
I imagine you are doing this to rerender on async work. You should do just that. Given that you also seem to agree that event handling needs to be task returning, we should just flow the errors out of the component into the renderer where they can be handled in a centralized place

Reply:
I'm not sure I really get what the suggestion is here. But given that we're changing the signature of this method is - I think it's going to be unimportant. Holla back if you still have feedback after I update this.